bundles.md

```diff
@@ -0,0 +1,133 @@
+# Attestation Bundle
+
+An attestation bundle is a collection of multiple attestations in a single file.
+This allows attestations from multiple different points in the software supply
+chain (e.g. Provenance, Code Review, Test Result, vuln scan, ...) to be grouped
+together, allowing users to make decisions based on all available information.
+
+**NOTE**: Attestation Bundles themselves are **not authenticated** instead each
+individual attestation is authenticated
+([using DSSE](https://github.com/secure-systems-lab/dsse)). As such, an attacker
+might be able to _delete_ an attestation without being detected.  Predicates that
+follow [the monotonic principle](spec/README.md#parsing-rules) should not have any
+issues with this behavior.
+
+## Data structure
+
+Attestation Bundles use [JSON Lines](https://jsonlines.org/) to store multiple
+[DSSEs](https://github.com/secure-systems-lab/dsse).
+
+*  Each line MUST contain a _single_ DSSE.
+*  Each line MAY contain any DSSE `payloadType`
+*  in-toto attestations (`payloadType` == `application/vnd.in-toto+json`) MAY use
+   any ` _type`/`predicateType`
+*  [in-toto Statements](spec/README.md#statement)
+   in a Bundle MAY reference different Subjects
+*  Consumers MUST ignore any attestations whose `payloadType`, `_type`, or `predicateType`
+   they do not understand.
+*  Attestations MAY be signed by different keys
+*  New attestations MAY be added to existing bundles
+*  Processing of a bundle MUST NOT depend on the order of the attestations.
+
+
+Example:
+
+```jsonl
+{ "payloadType": "application/vnd.in-toto+json", "payload": "...", "signatures": [...] }
+{ "payloadType": "application/vnd.in-toto+json", "payload": "...", "signatures": [...] }
+{ "payloadType": "application/vnd.foo+xml", "payload": "...", "signatures": [...] }
+```
+
+## File naming convention
+
+* Attestation Bundles SHOULD use the `.intoto.jsonl` extension.
+* Bundles that concern a single artifact SHOULD name the bundle file
+  `<artifact filename>.intoto.jsonl`.
+* Bundles that concern multiple artifacts SHOULD name the bundle file
+  `intoto.jsonl`.
```

Comment:
Why not just simplify and use one set of rules? Particular, the multiple artifacts case can cover single artifacts.

Reply:
We've seen cases where some builders emit multiple artifacts but want to emit separate attestations for each artifact.

E.g. They emit foo.out, foo.out.intoto.jsonl, and bar.out, bar.out.intoto.jsonl.

Reply:
The file names seem a bit awkward to me as well, specifically with the multiple artifacts rule. Also, does the last rule contradict the first one?

Reply:
Can you provide anything you're happier with?  FWIW .jsonl is the suggest JSON Lines suffix.

I can remove the first bullet.  I think it's redundant.

Reply:
> I can remove the first bullet. I think it's redundant.

I agree.

>  FWIW .jsonl is the suggest JSON Lines suffix.

I'm okay with the .jsonl extension. I wonder if we can use `<artifact name>.intoto.jsonl` and something like `multiple.intoto.jsonl` to remain consistent? I think that also handles the first bullet. Further, I'm guessing we're not considering `<user defined name>.intoto.jsonl` to allow for finding bundles in tooling? That is, "look for <artifact>.intoto.jsonl or just intoto.jsonl"-esque behaviour?

Apologies if these are silly questions! :)

Reply:
Removed first bullet.

I do like this idea of just calling it 'multiple.intoto.jsonl', for one thing this easily lets people grab all the extensions by just searching for .endswith('.intoto.jsonl') and not having to worry about 'notintoto.jsonl'.

Changed to 'multiple.intoto.jsonl'.  

> Further, I'm guessing we're not considering <user defined name>.intoto.jsonl to allow for finding bundles in tooling?

Right!

Going to resolve this comment, please reopen if there are still issues.

Reply:
No issues, thanks!